### PR TITLE
Use consistent temp folder path

### LINKED ISSUE #10657: Unable to download mods from browse online content (Download Denied, Temp folder isn't a allowed path)
##### Minetest version

```
5.3.0
```

##### OS / Hardware

Operating system: Windows 10

##### Summary
As the title says, I am unable to download mods/content from Minetest's "Browse online content". I have also tried to run Minetest as an administrator and still gives an error. I am however able to download the content from the website through a browser.
Log :
```
ERROR[AsyncWorker-0]: DOWNLOAD denied: I:\AppData\Temporary Files\Temp\sag4._MODNAME_mobs.zip isn't a allowed path
ERROR[AsyncWorker-0]: downloading "https://content.minetest.net/packages/TenPlus1/mobs/releases/5530/download/" failed
```
I had moved the Temporary Files folder's location from the default C-Drive to another drive in order to save space. However moving the folder back to the default location doesn't work and still gives me the error. Moving the Temp folder in the same drive as Minetest still gives the same error. As said earlier I have tried running the app as administrator and that didn't work. Other applications can access the folder normally.

##### Steps to reproduce
1. Open Minetest
2. Click on Browse online content
3. Click Install on any mod
4.  `An Error Occurred:`
   `Failed to Download (modname)`
And the logs are the same.

### PR DESCRIPTION
**Pull request by rubenwardy**
This should fix #10657, but **I'm unable to test on Windows**

## To do

- [x] Does `os.tmpname()` need to be replaced too?
- [ ] Test that this does fix #10657

## How to test

* Try installing from ContentDB

### CODE REVIEW

File: src/filesys.cpp [hunk 1]
```diff
@@ -359,8 +356,9 @@ std::string TempPath()
 		compatible with lua's os.tmpname which under the default
 		configuration hardcodes mkstemp("/tmp/lua_XXXXXX").
 	*/
+
 #ifdef __ANDROID__
-	return g_settings->get("TMPFolder");
+	return porting::path_cache;
 #else
 	return DIR_DELIM "tmp";
```
Comment (sfan5) [inline src/filesys.cpp:363]: this will 100% not work on Windows

Comment (rubenwardy) [reply src/filesys.cpp:363]: This is in the Linux section. Windows is elsewhere

File: builtin/mainmenu/common.lua [hunk 2]
```diff
@@ -146,35 +146,15 @@ end
...
+	local temp = core.get_temp_path()
+	return temp .. DIR_DELIM .. "MT_" .. math.random(0, 10000)
+end
 
-	local randname = "MTTempModFolder_" .. math.random(0,10000)
-	local backstring = filetocheck:reverse()
-	return filetocheck:sub(0, filetocheck:len() - backstring:find(DIR_DELIM) + 1) ..
-		randname
+--------------------------------------------------------------------------------
+os.tmpname = function()
```
Comment (rubenwardy) [inline builtin/mainmenu/common.lua:154]: Minetest doesn't actually use `os.tmpname` or `os.tmpfile`, so perhaps it's best not to override them

**Comment by rubenwardy** [comment]
I have tested this on Android

File: builtin/mainmenu/common.lua (hunk 2 above)
Comment (sfan5) [reply builtin/mainmenu/common.lua:154]: This is only in the mainmenu code so it's not like there are compatibility concerns.

**Comment by sfan5** [comment]
I cannot reproduce #10657 in my Windows 10 VM so I couldn't tell your if this _fixes_ it.
With this PR applies it still works however.

**Review by sfan5 — APPROVED** [review]
Tested (as you can see above)

**Pull request merged**